File: LeetCode Solutions/LeetCoding Challenge/June 2020 LeetCoding Challenge/Redundant Connection.py
class Solution:
    def findRedundantConnection(self, edges: List[List[int]]) -> List[int]:
        parentArray = [i for i in range(len(edges) + 1)]
        def find(x: int) -> int:
            if(x != parentArray[x]):
                parentArray[x] = find(parentArray[x])
            return parentArray[x]
        def union(x: int, y: int) -> None:
            parentArray[find(y)] = find(x)
        for a,b in edges:
            if(find(a) == find(b)):
                return [a,b]
            else:
                union(a,b)

File: LeetCode Solutions/LeetCoding Challenge/June 2020 LeetCoding Challenge/Redundant Connection.cpp
class Solution {
public:
    vector<int> findRedundantConnection(vector<vector<int>>& edges) {
        parent.resize(edges.size()+1);
        for(int i=0; i<parent.size(); i++){
            parent[i] = i;
        }
        for(auto& e : edges){
            if(find(e[0]) == find(e[1])){
                return e;
            }
            else{
                uniun(e[0], e[1]);
            }
        }
        return edges[0];
    }
private:
    vector<int> parent;
    int find(int x){
        if(x != parent[x]) parent[x] = find(parent[x]);
        return parent[x];
    }
    void uniun(int x, int y){
        parent[find(y)] = find(x);
    }
};